src/Compiler/pars.fsy

```diff
@@ -4663,13 +4666,29 @@ arrayExpr:
       { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())
         exprFromParseError ($2 (rhs2 parseState 1 2)) }
 
-  | LBRACK_BAR error BAR_RBRACK
-      { (* silent recovery *) SynExpr.ArrayOrList(true, [ ], lhs parseState) }
+  | LBRACK_BAR error BAR_RBRACK 
+      {  (* silent recovery *) SynExpr.ArrayOrList (CollectionType.Array, [ ], lhs parseState) }  
 
   | LBRACK_BAR recover
-      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())
+      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar())  
+        (* silent recovery *) 
+        exprFromParseError (SynExpr.ArrayOrList (CollectionType.Array, [ ], rhs parseState 1)) }  
+
+immarrayExpr:
+  | LBRACK_COLON immarrayExprElements COLON_RBRACK
+      {  $2 (lhs parseState) }
+
+  | LBRACK_COLON immarrayExprElements recover
+      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar()) // Don't know where this is, can't change to BracketColon
+        exprFromParseError ($2 (rhs2 parseState 1 2)) }
+
+  | LBRACK_COLON error COLON_RBRACK
+      {  (* silent recovery *) SynExpr.ArrayOrList (CollectionType.ImmutableArray, [ ], lhs parseState) }
+
+  | LBRACK_COLON recover
+      { reportParseErrorAt (rhs parseState 1) (FSComp.SR.parsUnmatchedBracketBar()) // Don't know where this is, can't change to BracketColon
```

Comment:
Don't know where this is, can't change to BracketColon

Reply:
> Don't know where this is, can't change to BracketColon

Is it from FSComp.txt? Maybe add the BracketColon message you need there?